### DISCUSSION TITLE
How to buy at close price (by default it only buys from open price)

### DISCUSSION THREAD

**Discussion by saetthakij** [kernc/backtesting.py#892] [Q&A] [answered]
How can I run a backtest by buying at close price (by default it only buys from open price)``

class wma(Strategy):

    wma_window = 2
    max_window = 10
    ma_window = 10

    # Do as much initial computation as possible
    def init(self):
        self.wma = self.I(ta.WMA, pd.Series(self.data.Close), self.wma_window)
        self.max = self.I(ta.MAX, pd.Series(self.data.Close), self.max_window)
        self.ma = self.I(ta.MA, pd.Series(self.data.Close), self.ma_window)
        self.close =  self.I(close_price,pd.Series(self.data.Close))

    def next(self):
        
        price = self.data.Close[-1]
        if crossover(self.data.Close, self.wma):
            self.position.close()
        elif crossover(self.wma, self.data.Close):
            self.buy()
        

bt = Backtest(df3, wma, cash=10_000, commission=.002)
stats = bt.run()
bt.plot()
print(stats)

**Comment by eervin123** [comment, marked as answer]
Take a look at the [quick start](https://kernc.github.io/backtesting.py/doc/examples/Quick%20Start%20User%20Guide.html) user guide. @kernc lays out the following if you want to trade on close 
<img width="1348" alt="image" src="https://user-images.githubusercontent.com/32274861/215837667-26e0fea4-19bc-4387-a428-b3d11a1c7c72.png">

**Reply by eervin123** [reply]
bt = Backtest(df3, wma, cash=10_000, commission=.002, trade_on_close=True)
